### PR TITLE
add unstable prop for disabling unregister entity

### PR DESCRIPTION
**Pull request by prasaria**
## Hey, I just made a Pull Request!

Add unstable props to disable unregister entity menu
Screen Shot 2021-07-12 at 16 32 13

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/small-roses-fly.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'example-app': patch
```
Comment (benjdlambert) [inline .changeset/small-roses-fly.md:2]: Don't need to bump the `example-app` as it's not a published package.

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 2]
```diff
@@ -124,7 +124,10 @@ const EntityLayoutWrapper = (props: { children?: ReactNode }) => {
 
   return (
     <>
-      <EntityLayout UNSTABLE_extraContextMenuItems={extraMenuItems}>
+      <EntityLayout
+        UNSTABLE_extraContextMenuItems={extraMenuItems}
+        UNSTABLE_contextMenuOptions={{ disableUnregister: true }}
```
Comment (benjdlambert) [inline packages/app/src/components/catalog/EntityPage.tsx:129]: Maybe let's not change the behaviour fo the `example-app` for contributors.

File: .changeset/small-roses-fly.md (hunk 1 above)
Comment (prasaria) [reply .changeset/small-roses-fly.md]: updated

File: packages/app/src/components/catalog/EntityPage.tsx (hunk 2 above)
Comment (prasaria) [reply packages/app/src/components/catalog/EntityPage.tsx]: updated

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Merged with the caveat that this is super unstable, and will possibly be removed or changed in future releases.

**Comment by alzafacon** [comment]
My team likes the ability to disable the unregister button. Here's our vote for keeping it. (even if it changes)

**Pull request merged**